ast/file.go

```diff
@@ -21,7 +21,6 @@ import "fmt"
 type FileDeclNode interface {
 	Node
 	Name() string
-	GetSyntax() Node
```

Comment:
While _technically_ a breaking change, I think it's _extremely_ unlikely that this will break any actual user code.

It turns out this method wasn't even used. And it feels out-of-place to have this and to _not_ have something for the editions node. So instead of adding a `GetEdition` or updating the docs/semantics of this method to return an `*EditionNode` if the file uses editions (but keep potentially misleading name), it seemed cleanest to just remove it.

Reply:
I don't think it's necessary. Usually when working with an AST, you'll have a concrete `*ast.FileNode` which already provides access to the syntax and edition AST nodes (via struct field access).

This interface is used in places in the compiler that want to report error locations, but do not have an AST. For cases where the AST is not available, the other impl (other than `*ast.FileNode`) returns a dummy source position that just has the name of the file, but not line and column (since that info isn't available). There's currently nothing in the compiler that needs to report anything about the syntax, so this method just isn't necessary.